src/WireMock.Net/Serialization/MappingConverter.cs

```diff
@@ -173,6 +187,49 @@ public string ToCSharpCode(IMapping mapping, MappingConverterSettings? settings
         return sb.ToString();
     }
 
+    private string ConvertJsonToAnonymousObjectDefinition(JToken token, int ind = 0)
```

Comment:
Another possible option would be to just use reflection:
``` c#
private static string GenerateObjectInitializationCodeRecursive(object anonymousObject)
{
	var properties = anonymousObject.GetType().GetProperties();
	var objectBuilder = new StringBuilder();

	objectBuilder.AppendLine("new");
	objectBuilder.AppendLine("{");

	foreach (var property in properties)
	{
		string propertyName = property.Name;
		object propertyValue = property.GetValue(anonymousObject);

		if (propertyValue != null)
		{
			if (propertyValue is string)
			{
				objectBuilder.AppendLine($"    {propertyName} = \"{propertyValue}\",");
			}
			else if (property.PropertyType.IsClass && !property.PropertyType.FullName.StartsWith("System."))
			{
				string nestedObjectCode = GenerateObjectInitializationCodeRecursive(propertyValue);
				objectBuilder.AppendLine($"    {propertyName} = {nestedObjectCode.Trim()},");
			}
			else
			{
				objectBuilder.AppendLine($"    {propertyName} = {propertyValue},");
			}
		}
	}

	objectBuilder.AppendLine("};");

	return objectBuilder.ToString();
}
```

Reply:
Yes, it's quite common that a given problem can be solved in multiple ways. Try to use the code from above and there are compilation issues (probably due to missing properties in some target frameworks) and I guess some additional adjustments are needed to achieve proper formatting.

Is there any reason why Newtonsoft.Json can't be used here? It's already referenced by WireMock project.

Reply:
Your solution is fine. It was just an idea.